src/graphforge/datasets/registry.py

```
@@ -0,0 +1,252 @@
+    Raises:
+        ValueError: If URL scheme is not HTTP or HTTPS
+        RuntimeError: If download fails
+    """
+    # Validate URL scheme for security (only allow HTTP/HTTPS)
+    from urllib.parse import urlparse
+
+    parsed = urlparse(url)
+    if parsed.scheme not in ("http", "https"):
+        raise ValueError(
+            f"Unsupported URL scheme '{parsed.scheme}'. Only HTTP and HTTPS are allowed."
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)